src/main/java/emailvalidator4j/parser/DomainPart.java

```diff
@@ -112,13 +113,25 @@ private boolean isDomainLiteral(boolean openBrackets) throws UnclosedDomainLiter
         return this.lexer.getCurrent().equals(Tokens.OPENBRACKET);
     }
 
-    private void checkNotAllowedChars(TokenInterface token) throws DomainNotAllowedCharacter {
+    private void checkNotAllowedChars(TokenInterface token) throws DomainNotAllowedCharacter, DomainHyphen {
         if (notAllowedTokens.contains(token)) {
             throw new DomainNotAllowedCharacter(
                     String.format("%s is not allowed in domain part", token.getName())
             );
         }
-
+        if (Tokens.GENERIC.equals(token.getName())) {
```

Comment:
Maybe the whole `if` block can be extracted to `private void checkNotAllowedCharsInSubDomain` ?

Reply:
Good idea! I’ll do that.